core/src/main/java/hudson/ClassicPluginStrategy.java

```diff
@@ -260,6 +260,40 @@ private static Manifest loadLinkedManifest(File archive) throws IOException {
                 createClassLoader(paths, dependencyLoader, atts), disableFile, dependencies, optionalDependencies);
     }
 
+    private static void fix(Attributes atts, List<PluginWrapper.Dependency> optionalDependencies) {
+        // don't fix the dependency for yourself, or else we'll have a cycle
```

Comment:
This comment might make more sense inside `getImpliedDependencies` where the `pluginName` is used.